src/config/syscheck-config.c

```diff
@@ -2504,37 +2504,57 @@ char* check_ascci_hex (char *input) {
 }
 
 static char **get_paths_from_env_variable (char *environment_variable) {
-
-    char **paths =NULL;
+    char **paths = NULL;
+    char *expandedpath = NULL;
+    int i = 0;
 
 #ifdef WIN32
-    char expandedpath[PATH_MAX + 1];
-
-    if(!ExpandEnvironmentStrings(environment_variable, expandedpath, PATH_MAX + 1)){
-        merror("Could not expand the environment variable %s (%ld)", expandedpath, GetLastError());
+    static const char *DELIM = ";";
+    DWORD env_var_size = ExpandEnvironmentStrings(environment_variable, NULL, 0);
+    if (env_var_size <= 0) {
+        merror(FIM_ERROR_EXPAND_ENV_VAR, environment_variable, GetLastError());
+        goto error;
     }
-
-    /* The env. variable may have multiples paths split by ; */
-    paths = OS_StrBreak(';', expandedpath, MAX_DIR_SIZE);
-
-    for (int i = 0; paths[i]; i++) {
-        str_lowercase(paths[i]);
+    os_calloc(env_var_size, sizeof(char), expandedpath);
+    if(!ExpandEnvironmentStrings(environment_variable, expandedpath, env_var_size)){
+        merror(FIM_ERROR_EXPAND_ENV_VAR, environment_variable, GetLastError());
+        os_free(expandedpath);
+        goto error;
     }
-
+    str_lowercase(expandedpath);
 #else
-    char *expandedpath = NULL;
-
+    static const char *DELIM = ":";
     if(environment_variable[0] == '$') {
         environment_variable++;
+        char *aux = getenv(environment_variable);
+        if (!aux) {
+            merror(FIM_ERROR_EXPAND_ENV_VAR, environment_variable);
+            goto error;
+        } else {
+            os_strdup(aux, expandedpath);
+        }
+    } else {
+        os_strdup(environment_variable, expandedpath);
     }
```

Comment:
If the parameter provided is not an environment variable, we should prevent any further processing on it here.
```suggestion
    static const char *DELIM = ":";
    if(environment_variable[0] != '$') {
        // not an environment variable.
        goto error;
    }
    environment_variable++;

    char *aux = getenv(environment_variable);
    if (!aux) {
        merror(FIM_ERROR_EXPAND_ENV_VAR, environment_variable);
        goto error;
    } else {
        os_strdup(aux, expandedpath);
    }
```

We should also reconsider renaming the `error` tag to something like `end`, since with the previous change it stops being an error condition exclusively.